File: Classic_Algorithms/collatz_conjecture/collatz_conjecture.go
package main

import (
	"fmt"
)

func main() {
  nums := []int{11}
  
	for {
    i := len(nums) - 1
    num := nums[i]
    if num == 1 {
      break
    }
    odd := num % 2 == 1
    if odd {
      nums = append(nums, 3 * num + 1)
    } else {
      nums = append(nums, num / 2)
    }
  }

  fmt.Println(nums)
}

File: Classic_Algorithms/collatz_conjecture/collatz_conjecture.rs
fn main() {
  let mut nums: Vec<i32> = vec![11];
  loop {
    let i = nums.len() - 1;
    let num = nums[i];
    if num == 1 {
      break;
    }
    let odd = num % 2 == 1;
    if odd {
      nums.push(3 * num + 1);
    } else {
      nums.push(num / 2);
    }
  }
  println!("{:?}", nums)
}
    
